dist_example.py

```diff
@@ -68,6 +81,27 @@ def train(
     _num_eg_print = -1
     _start_eg = copy(model._num_eg_processed)
 
+    # this value will be scaled 
+    n_workers = 4
+
+    # we will increase the number of workers when each is computing more than
+    # `scale_threashold` gradients
+    # scale factor determines by how much we will the number of workers upon reaching the threshold
+    scale_threshold = 16
+    scale_factor = 1.25;
+    cluster = LocalCluster(n_workers=n_workers)
+
+    # dask support adaptive scaling, the docs say this is optimized / designed for interactive work loads that
+    # may be idle for long amounts of time, so I am not sure that this is the best solution. I really like the
+    # more explicit approach
+    # cluster.adapt(minimum=0, maximum_memory="100 GiB", interval='500ms')
+
+    # using dasks built in serializer causes issue
+    client = Client(cluster, serializers=['pickle'])
+    future_inputs = client.scatter(inputs)
+    future_targets = client.scatter(targets)
+    future_device = client.scatter(device)
```

Comment:
Is this necessary to scatter? `device` should be small.

``` 
[ins] In [18]: import pickle
[ins] In [19]: import torch
[ins] In [20]: len(pickle.dumps(torch.device("cpu")))
Out[20]: 35
```

Looks like it's only 35 bytes.